airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -136,6 +142,66 @@ def test_two_sequential_reads(
                 record_value, state_value, threshold_days
             ), f"Second incremental sync should produce records older or equal to cursor value from the state. Stream: {stream_name}"
 
+    def test_read_sequential_slices(
+        self, inputs: IncrementalConfig, connector_config, configured_catalog_for_incremental, cursor_paths, docker_runner: ConnectorRunner
+    ):
+        """
+        Incremental test that makes an initial API request without a state checkpoint. Then we partition the results by stream and
+        slice checkpoints resulting in batches of messages that look like:
+        <state message>
+        <record message>
+        ...
+        <record message>
+
+        Using these batches, we then make API requests using the state message and verify the the correctness of the messages in the
+        response.
```

Comment:
Nice description! I'm not sure it's appropriate to talk about "APIs" here though. Maybe say "call the read method" instead of "make an API request"?